src/components/implementation/valloc/simple/valloc.c

```diff
@@ -114,8 +165,8 @@ int valloc_alloc_at(spdid_t spdid, spdid_t dest, void *addr, unsigned long npage
 		    !(trac = cos_vect_lookup(&spd_vect, dest))) goto done;
 	}
 
-	if (unlikely(npages > MAP_MAX * 32)) {
-		printc("valloc: cannot alloc more than %u bytes in one time!\n", 32 * WORDS_PER_PAGE * PAGE_SIZE);
+	if (unlikely(npages > MAP_MAX * sizeof(u32_t))) {
```

Comment:
this is not correct. 
MAP_MAX   = WORDS_PER_PAGE
32                = BITS_PER_WORD
sizeof(u32_t) = BYTES_PER_WORD
Here we caculate the total number of bits, not bytes, in a page